bun/whatsapp-with-vonage/src/main.ts

```diff
@@ -0,0 +1,67 @@
+import jwt from "jsonwebtoken";
+import { fetch } from "undici";
+
+export default async ({ req, res, log, error }) => {
+  if (req.method === "GET") {
+    const fileName = Bun.resolveSync("../static/index.html", import.meta.dir);
+    const html = await Bun.file(fileName).text();
+
+    return res.send(html, 200, {
+      "Content-Type": "text/html; charset=utf-8",
+    });
+  }
+
+  try {
+    const hasher = new Bun.CryptoHasher("sha256");
+    const hashedBuffer = hasher.update(JSON.stringify(req.body)).digest("hex");
+    const hashedValue = hashedBuffer.toString();
+
+    const authHeader = req.headers.authorization.split(" ")[1];
+    const decodedToken = jwt.verify(
+      authHeader,
+      Bun.env.VONAGE_SIGNATURE_SECRET,
+      {
+        algorithms: ["HS256"],
+      }
+    );
+    if (hashedValue != decodedToken["payload_hash"]) {
+      return res.json({
+        ok: false,
+        error: "Payload mismatched",
+      });
+    }
+  } catch (err) {
+    return res.json({
+      ok: false,
+      error: "can't verify",
+    });
+  }
+
+  const secret = `${Bun.env.VONAGE_API_KEY}:${Bun.env.VONAGE_ACCOUNT_SECRET}`;
+  const basicAuthToken = btoa(secret);
+
+  if (!(req.body.text == null)) {
+    await fetch(`https://messages-sandbox.nexmo.com/v1/messages`, {
+      method: "POST",
+      body: JSON.stringify({
+        from: Bun.env.VONAGE_WHATSAPP_NUMBER,
+        to: req.body.from,
+        message_type: "text",
+        text: `You sent me: ${req.body.text}`,
+        channel: "whatsapp",
+      }),
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Basic ${basicAuthToken}`,
+      },
+    });
+  }
```

Comment:
Swap the case round and use early return here please https://refactoring.guru/replace-nested-conditional-with-guard-clauses